### DISCUSSION TITLE
Tip: coloring confidence scores

### DISCUSSION THREAD

**Discussion by jsalsman** [deepgram/community#472] [General help] [answered]
If you cube confidence scores and then convert them to hue degrees multiplying by 120, you can use maximum saturation and value to make background colors from green (1.0) to red (0.0) which are high contrast behind black text (avoiding dark browns in the linear mapping between green and red.) In Python:
```
def confidenceColor(confidence):  # High contrast color mapping for HTML 'rgb(r,g,0)'
  # Hue: Red (0) to Green (120)
  hue = 120 * confidence**3  # cube so green falls off faster
  # Conversion from HSV to RGB where saturation and value are both 100%
  x = 1 - abs((hue / 60.0) % 2 - 1)
  r, g = (1, x) if hue < 60 else (x, 1)
  return f'rgb({int(r*255)},{int(g*255)},0)'
```
Not only does cubing the scores make them seem more in line with general intelligibility to my ear (and eye), but also, everything below 0.6 is red, somewhat analogous to the standard classroom letter grading scale (in the US anyway). This is with nova-2 streaming.

It's really useful to color transcriptions. For example, reviewers can immediately see areas of potential mistakes. Enjoy!

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jsalsman** [comment, marked as answer]
Here's what it looks like:

![Screenshot 2023-12-02 1 24 49 AM](https://github.com/deepgram/community/assets/3393748/051bac52-f6fc-45e6-ad8d-4a4ef0a5dab5)

See https://arxiv.org/pdf/2204.04288.pdf for more information.
